### DISCUSSION TITLE
Is it possible to enforce a max length on a string with a mapping?

### DISCUSSION THREAD

**Discussion by robfolks** [JoshClose/CsvHelper#2110] [Q&A] [answered]
Hi,

I'm importing a string into a SQL table with a max length of 50. Is it possible to enforce a max length on a string value in one column with a mapping? If the string being imported is longer than 50 characters I would want to import the first 50 characters.

**Comment by JoshClose** [comment, marked as answer]
No. You could write a custom converter to do it though.

```cs
public class TruncateStringConverter : StringConverter
{
	public override object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
	{
		var s = base.ConvertFromString(text, row, memberMapData) as string;
		return s == null ? s : s.Substring(0, 50);
	}
}
```

You can apply that via a mapping like

```cs
Map(m => m.StringProperty).TypeConverter<TruncateStringConverter>();
```

Or globally for all strings like

```cs
csv.Context.TypeConverterCache.AddConverter<string>(new TruncateStringConverter());
```

**Reply by robfolks** [reply]
Thanks for the advice. 

For anyone else who might read this I tweaked the convert method slightly because the string doesn't always exceed 50 characters. 

>         public override object ConvertFromString(string text, IReaderRow row, MemberMapData memberMapData)
>        {
>           var s = base.ConvertFromString(text, row, memberMapData) as string;
>           if (string.IsNullOrEmpty(s)) return s;
>           return s.Length <= 50 ? s : s.Substring(0, 50);
>        }
